backend-rs/src/search.rs

```diff
@@ -0,0 +1,67 @@
+use opensearch::{
+    http::StatusCode,
+    indices::{IndicesCreateParts, IndicesGetParts},
+    OpenSearch, SearchParts,
+};
+use serde_json::{json, Value};
+use std::collections::HashMap;
+
+pub async fn create_flake_index(opensearch: &OpenSearch) -> Result<(), opensearch::Error> {
+    let status = opensearch
+        .indices()
+        .get(IndicesGetParts::Index(&["flakes"]))
+        .send()
+        .await?
+        .status_code();
+
+    if status == StatusCode::NOT_FOUND {
+        let _ = opensearch
+            .indices()
+            .create(IndicesCreateParts::Index("flakes"))
+            .send()
+            .await?;
+    }
```

Comment:
Check if opensearch index already exists. 

Turns out the opensearch crate is a wrapper around http calls, and trying to get information on a non-existing index returns a 404